pygfx/utils/transform.py

```diff
@@ -544,6 +569,9 @@ def flag_update(self):
     @property
     def matrix(self):
         view_array = self.untracked_matrix.view()
+        # Stop the user from accidentally writing the temporary arrays
+        # that we return from these operations
+        # https://github.com/pygfx/pygfx/issues/651
         view_array.flags.writeable = False
```

Comment:
Does this flag only the view or does it also flag the original array?

Reply:
only the view:

```python
In [1]: import numpy as np

In [2]: a = np.zeros((2, 3))

In [3]: b = a.view()

In [4]: b.flags.writeable = False

In [5]: assert a.flags.writeable

In [6]: a.flags.writeable
True

In [7]: assert not b.flags.writeable

In [8]: b.flags.writeable
False
```